src/main/java/org/folio/innreach/domain/service/impl/CirculationServiceImpl.java

```diff
@@ -258,13 +262,46 @@ public InnReachResponseDTO returnUncirculated(String trackingId, String centralC
 
     if (state == ITEM_RECEIVED || state == RECEIVE_UNANNOUNCED) {
       transaction.setState(RETURN_UNCIRCULATED);
-      transactionRepository.save(transaction);
       return success();
     } else {
       throw new IllegalArgumentException("Transaction state is not " + ITEM_RECEIVED.name() + " or " + RECEIVE_UNANNOUNCED.name());
     }
   }
 
+  @Override
+  public InnReachResponseDTO recall(String trackingId, String centralCode, RecallDTO recallDTO) {
+    var transaction = getTransaction(trackingId, centralCode);
+    var requestId = transaction.getHold().getFolioRequestId();
+    var request = requestService.findRequest(requestId);
+    var requestStatus = request.getStatus();
+
+    if (requestStatus == OPEN_AWAITING_PICKUP || requestStatus == OPEN_IN_TRANSIT) {
+      try {
+        requestService.cancelRequest(transaction, "Item has been recalled.");
+      } catch (Exception e) {
+        throw new CirculationException("Unable to create a cancel request on the item: " + e.getMessage());
```

Comment:
`throw new CirculationException("Unable to create a cancel request on the item: " + e.getMessage(), e);`